File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.cc
#include <bits/stdc++.h>
using namespace std;

string s;

bool includeV(int idx) {
    return (idx == 'a' || idx == 'e' || idx == 'i' || idx == 'o' || idx == 'u');
}

int main() {
    ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

    while(true) {
        cin >> s;
        if (s == "end") break;
        
        int vcnt = 0, ccnt = 0, prev = -1;
        bool flag = 0, include_v = 0;

        for (int i=0; i<s.size(); i++) {
            // 조건 1
            int idx = s[i];
            if (includeV(idx)) vcnt++, ccnt = 0, include_v = 1;
            else vcnt = 0, ccnt++;

            // 조건 2
            if (vcnt == 3 || ccnt == 3) flag = 1;

            // 조건 3
            if (i >= 1 && (prev == idx) && (idx != 'e' && idx != 'o')) flag = 1;
            prev = idx;
        }
        
        if (!include_v) flag = 1;
        if (flag) cout << '<' << s << "> is not acceptable.\n";
        else cout << '<' << s << "> is acceptable.\n";
    }
    return 0;
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.py
import sys
input = lambda: sys.stdin.readline().rstrip()

def check(string):
    prev = string[0]
    includeV, flag = False, True
    ccnt, vcnt = 0, 0

    for idx, s in enumerate(string):
        if s in "aeiou":
            includeV = True
            vcnt += 1
            ccnt = 0
        else:
            ccnt += 1
            vcnt = 0

        if ccnt == 3 or vcnt == 3:
            flag = False

        if idx >= 1 and prev == s and (s != "e" and s != "o"):
            flag = False
        
        prev = s

    if not includeV: return False
    if not flag: return False
    return True

while True:
    s = input()
    if s == "end":
        break
    
    ret = check(s)
    if ret:
        print(f"<{s}> is acceptable.")
    else:
        print(f"<{s}> is not acceptable.")